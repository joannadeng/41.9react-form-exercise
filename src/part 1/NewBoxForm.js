import React, { useState } from "react";


const NewBoxForm = ({addBox}) => {
    const initialState = {
        bgcolor:" ",
        height:" ",
        width:" "
    }

    const [boxEle, setBoxEle] = useState(initialState);

    const handleChange = (e) => {
        const {name,value} = e.target;
        if(!isNaN(value)) {
            setBoxEle(boxEle => ({
            ...boxEle,
            [name]:+value
            }))
        }else{
            setBoxEle(boxEle => ({
                ...boxEle,
                [name]:value
                }))
        } 
    }


    const handleSubmit =(e) => {
        e.preventDefault();
        addBox({...boxEle});
        setBoxEle(initialState);  // why cant reset ?
    }

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="bgcolor">Background Color</label>
            <input 
            id="bgcolor"
            name="bgcolor"
            value={boxEle.value}
            type="text"
            placeholder="background color"
            onChange={handleChange}
            />

            <label htmlFor="width">Width</label>
            <input 
            id="width"
            name="width"
            value={boxEle.value}
            type="number"
            min="30"
            placeholder="width"
            onChange={handleChange}
            />

            <label htmlFor="height">Height</label>
            <input 
            id="height"
            name="height"
            value={boxEle.value}
            type="number"
            min="30"
            placeholder="height"
            onChange={handleChange}
            />
            <button>Submit!</button>
        </form>

    )
}

export default NewBoxForm;