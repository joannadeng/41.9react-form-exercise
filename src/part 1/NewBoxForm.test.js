import {render} from '@testing-library/react';
import NewBoxForm from './NewBoxForm';

it ('it should render without crashing', () => {
    render(<NewBoxForm />)
})

it ("should match snapshot", () => {
    const {asFragment} = render(<NewBoxForm />);
    expect(asFragment()).toMatchSnapshot();
})