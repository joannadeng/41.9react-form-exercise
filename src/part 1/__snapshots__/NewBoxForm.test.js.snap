// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="bgcolor"
    >
      Background Color
    </label>
    <input
      id="bgcolor"
      name="bgcolor"
      placeholder="background color"
      type="text"
      value=""
    />
    <label
      for="width"
    >
      Width
    </label>
    <input
      id="width"
      min="30"
      name="width"
      placeholder="width"
      type="number"
      value=""
    />
    <label
      for="height"
    >
      Height
    </label>
    <input
      id="height"
      min="30"
      name="height"
      placeholder="height"
      type="number"
      value=""
    />
    <button>
      Submit!
    </button>
  </form>
</DocumentFragment>
`;
